bin/create_tls_key.sh

```diff
@@ -1,7 +1,21 @@
 #!/bin/bash
-set -xe
+set -e
 
-mkdir -p ~/.irslackd
-openssl req -newkey rsa:4096 -nodes -sha512 -x509 -days 3650 -nodes -out ~/.irslackd/cert.pem -keyout ~/.irslackd/pkey.pem
-fingerprint=$(openssl x509 -noout -fingerprint -sha512 -inform pem -in ~/.irslackd/cert.pem | cut -d= -f2-)
+while getopts ":d:" opt; do
+  case $opt in
+    d) dir="$OPTARG"
+    ;;
+    \?) echo "Invalid option -$OPTARG" >&2
+    ;;
+  esac
+done
+if [ -z "$dir" ]; then
+   dir="~/.irslackd"
+fi
+dir=$(eval echo $dir)
```

Comment:
I presume this is to realpath or shell expand the directory. Any reason why we need to do that?

Reply:
That's because when using the `$dir` variable starting with a `~` (like the default directory) without using `eval`, instead of creating the folder `.irslackd` inside the $HOME directory, the folder `.irslackd` was created inside a folder named `~` in the current directory.

With eval: `/home/user/.irslackd`

Without eval: `<currentDir>/~/.irslackd`

Reply:
That's because without the `eval` the `~` would be interpreted as a folder name and not the `$HOME` directory.

Exmaple for the default directory `~/.irslackd`.

With eval: `/home/user/.irslackd`

Without eval: `<currentDir>/~/.irslackd`

Reply:
Ah, good call! Thanks for the patch!